crates/stackable-operator/src/cli.rs

```diff
@@ -205,12 +206,20 @@ pub struct ProductOperatorRun {
     /// Provides the path to a product-config file
     #[arg(long, short = 'p', value_name = "FILE", default_value = "", env)]
     pub product_config: ProductConfigPath,
+
     /// Provides a specific namespace to watch (instead of watching all namespaces)
     #[arg(long, env, default_value = "")]
     pub watch_namespace: WatchNamespace,
+
     /// Tracing log collector system
     #[arg(long, env, default_value_t, value_enum)]
     pub tracing_target: TracingTarget,
+
+    /// Kubernetes cluster domain, usually this is `cluster.local`.
+    // We are not using a default value here, as operators will probably do an more advanced
+    // auto-detection of the cluster domain in case it is not specified in the future.
+    #[arg(long, env)]
+    pub kubernetes_cluster_domain: Option<DomainName>,
```

Comment:
Maybe we should break this out into an opaque (to the operator) `KubernetesClusterInfoOpts` (or whatever) param, so that we can isolate the exact contract of how this works to `cluster_info`, and avoid having to do all the config plumbing again when this logic changes.

Reply:
For example, I could see us likely ending up adding a `--discovery-plugins=Kubeadm,K3s,ResolvConf` param.

Reply:
Makes total sense, ce51bc5940cd63bc4047afc3955b73a80030156a